wikipedia_cleanup/random_forest.py

```diff
@@ -89,42 +90,31 @@ def _fit_classifier(
             self.regressors[key] = reg
             self.last_preds[key] = (DUMMY_TIMESTAMP, 0)
 
-    def predict_timeframe(
-        self,
-        data_key: np.ndarray,
-        additional_data: np.ndarray,
-        columns: List[str],
-        first_day_to_predict: date,
-        timeframe: int,
-    ) -> bool:
-        if len(data_key) == 0:
-            return False
-        key_column_idx = columns.index("key")
-        data_key_item = data_key[0, key_column_idx]
-        if data_key_item not in self.regressors:
-            # checks if model has been trained for the key
-            # (it didn't if there was no train data)
-            return False
+    def _predict_next_change(
+        self, data_key: np.ndarray, columns: List[str]
+    ) -> datetime:
 
         value_valid_from_column_idx = columns.index("value_valid_from")
         sample = data_key[-1, ...]
         sample_value_valid_from = sample[value_valid_from_column_idx]
-        if self.last_preds[data_key_item][0] != sample_value_valid_from:
-            reg = self.regressors[data_key_item]
-            indices = [
-                columns.index(attr)
-                for attr in self.get_relevant_attributes()
-                if not (attr == "value_valid_from" or attr == "days_until_next_change")
-            ]
-            X_test = sample[indices].reshape(1, -1)
-            pred = int(reg.predict(X_test)[0])
-            self.last_preds[data_key_item] = (sample_value_valid_from, pred)
+        key_column_idx = columns.index("key")
+        current_key = data_key[0, key_column_idx]
 
-        else:
-            pred = self.last_preds[data_key_item][1]
+        reg = self.regressors[current_key]
```

Comment:
You accidentally removed the cache and return a Day instead. I guess the second one is intended, though I do not get why

Reply:
The cache gets added via the `RegressionPredictor`. The other one is to make it uniform with the other RegressionPredictors